src/presentation/controller/addressLookUp/Controller.ts

```diff
@@ -461,91 +461,85 @@ class AddressLookUpController extends AbstractController {
 
   generalPartnerTerritoryChoice() {
     return (request: Request, response: Response, next: NextFunction) => {
-      try {
-        const { ids, pageType } = super.extract(request);
-        const parameter = request.body.parameter;
-        const pageRouting = super.getRouting(addressLookUpRouting, pageType, request);
-
-        const isUnitedKingdom = parameter === "unitedKingdom";
-        const isGeneralPartnerURAPage =
-          pageType === AddressLookUpPageType.territoryChoiceGeneralPartnerUsualResidentialAddress;
-        const isGeneralPartnerPOAPage =
-          pageType === AddressLookUpPageType.territoryChoiceGeneralPartnerPrincipalOfficeAddress;
-        const isGeneralPartnerCorrespondenceAddressPage =
-          pageType === AddressLookUpPageType.territoryChoiceGeneralPartnerCorrespondenceAddress;
-
-        let redirectUrl;
-
-        if (isGeneralPartnerURAPage) {
-          redirectUrl = isUnitedKingdom
-            ? POSTCODE_GENERAL_PARTNER_USUAL_RESIDENTIAL_ADDRESS_URL
-            : ENTER_GENERAL_PARTNER_USUAL_RESIDENTIAL_ADDRESS_URL;
-        } else if (isGeneralPartnerPOAPage) {
-          redirectUrl = isUnitedKingdom
-            ? POSTCODE_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL
-            : ENTER_GENERAL_PARTNER_PRINCIPAL_OFFICE_ADDRESS_URL;
-        } else if (isGeneralPartnerCorrespondenceAddressPage) {
-          redirectUrl = isUnitedKingdom
-            ? POSTCODE_GENERAL_PARTNER_CORRESPONDENCE_ADDRESS_URL
-            : ENTER_GENERAL_PARTNER_CORRESPONDENCE_ADDRESS_URL;
-        }
-
-        redirectUrl = super.insertIdsInUrl(redirectUrl, ids.transactionId, ids.submissionId, ids.generalPartnerId);
-
-        const cacheKey = pageRouting.data?.[AddressCacheKeys.territoryCacheKey];
-
-        if (cacheKey) {
-          const cache = this.cacheService.addDataToCache(request.signedCookies, {
-            [ids.transactionId]: {
-              [cacheKey]: parameter
-            }
-          });
-          response.cookie(APPLICATION_CACHE_KEY, cache, cookieOptions);
+      this.handleTerritoryChoice(
+        request,
+        response,
+        next,
+        {
```

Comment:
might be better to somehow add the ukUrl and nonUkUrl to the pageRouting, maybe in data:

Reply:
I've now aded a nextUrlOverseas to the data in page routing. nextUrl will be used for uk choice.